src/ingest/src/main/scala/osmesa/ingest/util/Caching.scala

```diff
@@ -25,25 +25,23 @@ class S3Caching(cacheDir: String) extends Caching {
     s3client.doesObjectExist(bucket, prefix(filename))
 
   def orc(filename: String)(sparkjob: => DataFrame)(implicit ss: SparkSession): DataFrame = {
-    if (s3exists(filename)) {
-      ss.read.orc(fileUri(filename))
-    } else {
+    if (!s3exists(filename)) {
       sparkjob.repartition(1).write.format("orc").save(fileUri(filename))
-      sparkjob
     }
+
+    ss.read.orc(fileUri(filename))
```

Comment:
This re-reads the saved sparkjob, when we could just pass it back after saving it off, e.g. this would make any cached shuffle blocks still valid, while the read would lose them.

Reply:
No disk or memory issues. I'll remove that commit and keep an eye out.

Reply:
To be sure, the previous bit of code doesn't save off the DataFrame if the file doesn't exist. Do we want that behavior, so that the next time around it'll just pull from the ORC file? If so, we should keep that change, and just return the `sparkJob` instead of a newly read in copy.

Reply:
I think we want the previous behavior (now the current behavior in this PR).